File: C#/MigratoryBirds.cs
public static int migratoryBirds(List<int> arr)
{
    Dictionary<int,int> birdCountListDict = new Dictionary<int,int>();
    int mostFrequentBirdCount = 0;
    List<int> sameFrequentList = new List<int>();
    
    foreach(int bird in arr){
        if(birdCountListDict.ContainsKey(bird)){
            
            birdCountListDict[bird] =  birdCountListDict[bird] + 1;
            if( birdCountListDict[bird] > mostFrequentBirdCount ){
                
                mostFrequentBirdCount = birdCountListDict[bird];
                sameFrequentList  = new List<int>();
                sameFrequentList.Add(bird);
            }else if(birdCountListDict[bird] == mostFrequentBirdCount ){
                sameFrequentList.Add(bird);
            }
        }else{
            birdCountListDict[bird]=1;
        }
    }
    return sameFrequentList.Min();
}

File: Python/MigratoryBirds.py
def migratoryBirds(arr):
    return max(set(arr), key = arr.count)

#dictionary version runs fastly

def migratoryBirds(arr):
    birdDict = {}
    mostFrequentBird=""
    mostFrequentCount=0
    sameFrequentBirdList=[]
    for bird in arr:
        birdDict[bird] = birdDict.get(bird,0) + 1
        if(birdDict[bird] > mostFrequentCount):
            mostFrequentCount = birdDict[bird]
            mostFrequentBird = bird
            sameFrequentBirdList=[]
            sameFrequentBirdList.append(bird)
        elif (birdDict[bird] == mostFrequentCount):
            sameFrequentBirdList.append(bird)
    return min(sameFrequentBirdList)